pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go

```diff
@@ -1344,20 +1350,25 @@ var _ = Describe("Validating VMICreate Admitter", func() {
 			}),
 			Entry("with empty DNSPolicy", nil, nil),
 		)
-
 		DescribeTable("Should reject invalid DNSPolicy and DNSConfig",
 			func(dnsPolicy k8sv1.DNSPolicy, dnsConfig *k8sv1.PodDNSConfig, causeCount int, causeMessage []string) {
-				vmi := api.NewMinimalVMI("testvmi")
-				vmi.Spec.DNSPolicy = dnsPolicy
-				vmi.Spec.DNSConfig = dnsConfig
-				causes := ValidateVirtualMachineInstanceSpec(k8sfield.NewPath("fake"), &vmi.Spec, config)
-				Expect(causes).To(HaveLen(causeCount))
+				vmi := newBaseVmi(
+					withDNSPolicy(dnsPolicy),
+					withDNSConfig(dnsConfig),
+				)
+
+				ar, err := newAdmissionReviewForVMICreation(vmi)
+				Expect(err).ToNot(HaveOccurred())
+
+				resp := vmiCreateAdmitter.Admit(context.Background(), ar)
+				Expect(resp.Allowed).To(BeFalse())
+				Expect(resp.Result.Details.Causes).To(HaveLen(causeCount))
 				for i := 0; i < causeCount; i++ {
-					Expect(causes[i].Message).To(Equal(causeMessage[i]))
+					Expect(resp.Result.Details.Causes[i].Message).To(Equal(causeMessage[i]))
 				}
 			},
 			Entry("with invalid DNSPolicy FakePolicy", k8sv1.DNSPolicy("FakePolicy"), &k8sv1.PodDNSConfig{}, 1,
-				[]string{"DNSPolicy: FakePolicy is not supported, valid values: [ClusterFirstWithHostNet ClusterFirst Default None ]"}),
+				[]string{"spec.dnsPolicy in body should be one of [ClusterFirst ClusterFirstWithHostNet Default None]"}),
```

Comment:
The previous test was using `ValidateVirtualMachineInstanceSpec` meaning it would bypass Admit's `webhookutils.ValidateSchema` [1] and it fails as it's an invalid option when validated against the valid options of that field[2][3].

[1]https://github.com/kubevirt/kubevirt/blob/7915db1f7b18d30db3881a8a51d44abd0f3f76b4/pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go#L101
[2]https://github.com/kubevirt/kubevirt/blob/7915db1f7b18d30db3881a8a51d44abd0f3f76b4/vendor/github.com/go-openapi/validate/validator.go#L172
[3]https://github.com/kubevirt/kubevirt/blob/7915db1f7b18d30db3881a8a51d44abd0f3f76b4/staging/src/kubevirt.io/client-go/api/openapi_generated.go#L9329-L9336

Would you like me to add this to the commit message?

Reply:
Yes, please.
Thank you for explaining.